pkg/apis/keepalive/v1beta1/register.go

```diff
@@ -0,0 +1,92 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// VirtualIP is a keepalived CRD specificiation for virtual IP addresses
+type VirtualIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   VirtualIPSpec `json:"spec"`
+	Status Status        `json:"status"`
+}
+
+// VirtualIPSpec defines the spec of the CRD in a vrrp_instance section
+type VirtualIPSpec struct {
+	// Name defines the name of the vrrp_instance
+	Name string `json:"name"`
+
+	// IP virtual IP to use
+	IP string `json:"ip"`
+
+	// Interface interface where the virtual IP should be configured
+	// The default value is eth0
+	Interface *string `json:"interface,omitempty"`
```

Comment:
May be better to detect by CIDR on machine with keepalived controller?

And then may be webhook, to check if interface exists?